app/src/main/java/net/squanchy/tweets/view/TweetsAdapter.java

```diff
@@ -8,42 +8,66 @@
 
 import com.twitter.sdk.android.core.models.Tweet;
 import com.twitter.sdk.android.tweetui.Timeline;
-import com.twitter.sdk.android.tweetui.internal.TimelineDelegate;
 
 import net.squanchy.R;
 import net.squanchy.tweets.TweetsPageView;
 
-public class TweetsAdapter extends RecyclerView.Adapter<TweetViewHolder> {
+public class TweetsAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
 
     private final Context context;
-    private final TimelineDelegate<Tweet> delegate;
+    private final TwitterRepository repository;
 
     public TweetsAdapter(Timeline<Tweet> timeline, Context context) {
         this.context = context;
-        this.delegate = new TimelineDelegate<>(timeline);
+        this.repository = new TwitterRepository(timeline);
```

Comment:
can you have the repository as a dependency here instead of the timeline? :ghost: 

On another note: I don't think the repository should be inside the adapter, the adapter should just _receive_ data (i.e. with an `updateWith(xxx)` method) and adapt it for the view, the repository should be used and then the results fed into the adapter IMHO.